File: JS/Tree/PostOrderTraversalIter.js
// Definition for a  binary tree node
//			function TreeNode(data){
//				this.data = data
//				this.left = null
//				this.right = null
//			}

module.exports = {
	//param root : root node of tree
	//return a array of integers
	postorderTraversal : function(root){
        var res = [], stack = [];
        while(root != null){
            res.push(root.data);
            if(root.left != 'undefined' && root.left != null)stack.push(root.left);
            if(root.right != 'undefined' && root.right != null)stack.push(root.right);
            root = stack.pop();
        }
        res.reverse();
        return res;
	}
};

File: Java/Trees/PostorderTraversalIter.java
/**
 * Definition for binary tree
 * class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
	public ArrayList<Integer> postorderTraversal(TreeNode a) {
	    ArrayList<Integer> res = new ArrayList<Integer>();
	    if(a == null)  return res;
	    // Do root right left, that's very easy
	    // and then reverse the arraylist
	    Stack<TreeNode> stack = new Stack<>();
	    stack.push(a);
	    while(!stack.isEmpty()){
	        a = stack.pop();
	        res.add(a.val);
	        // for root, right, left. add left to stack first
	        if(a.left != null)stack.push(a.left);
	        if(a.right != null)stack.push(a.right);
	    }
	    Collections.reverse(res);
	    return res;
	}

}
